apps/finance/app/src/components/Transfers.js

```diff
@@ -93,64 +136,28 @@ const Transfers = React.memo(({ tokens, transactions }) => {
     setSelectedToken(0)
     setSelectedTransferType(0)
   })
-  const getCsvFilename = React.useCallback(() => {
-    let filename = 'transfers.csv'
-    const { start, end } = selectedDateRange
-    if (start && end) {
-      const formattedStart = format(selectedDateRange.start, 'yyyy-MM-dd')
-      const formattedEnd = format(selectedDateRange.end, 'yyyy-MM-dd')
-      filename = `transfers_${formattedStart}_to_${formattedEnd}.csv`
-    }
-    return filename
-  })
   const showMoreTransfers = React.useCallback(() => {
     setDisplayedTransfers(displayedTransfers + INITIAL_TRANSFERS_PER_PAGE)
   })
   const filteredTransfers = getFilteredTransfers({
-    tokens,
     transactions,
-    selectedToken,
+    selectedToken: selectedToken !== 0 ? tokens[selectedToken - 1] : null,
     selectedTransferType,
     selectedDateRange,
   })
   const symbols = tokens.map(({ symbol }) => symbol)
   const tokenDetails = tokens.reduce(getTokenDetails, {})
   const filtersActive = selectedToken !== 0 || selectedTransferType !== 0
   const { resolve: resolveAddress } = React.useContext(IdentityContext)
-  const getDownloadData = async () => {
-    const mappedData = await Promise.all(
-      filteredTransfers.map(
-        async ({
-          date,
-          numData: { amount },
-          reference,
-          isIncoming,
-          entity,
-          token,
-        }) => {
-          const { symbol, decimals } = tokenDetails[toChecksumAddress(token)]
-          const formattedAmount = formatTokenAmount(
-            amount,
-            isIncoming,
-            decimals,
-            true,
-            { rounding: 5 }
-          )
-          const { name = '' } = (await resolveAddress(entity)) || {}
-          return `${formatDate(
-            date
-          )},${name},${entity},${reference},${`${formattedAmount} ${symbol}`}`
-        }
-      )
+  const handleDownload = React.useCallback(async () => {
+    const data = await getDownloadData(
+      filteredTransfers,
+      tokenDetails,
+      resolveAddress
     )
-    const csvContent = [
-      'data:text/csv;charset=utf-8,Date,Name,Source/Recipient,Reference,Amount',
-    ]
-      .concat(mappedData)
-      .join('\n')
-
-    return window.encodeURI(csvContent)
-  }
+    const filename = getDownloadFilename(dao, selectedDateRange)
+    download(data, filename, 'data:text/csv')
+  })
```

Comment:
Should we add `filteredTransfers`, `tokenDetails`, and `resolveAddress` to this hook's cached list?